src/debug_utils-inl.h

```diff
@@ -27,13 +27,38 @@ struct ToStringHelper {
   }
   static std::string Convert(const std::string& value) { return value; }
   static std::string Convert(bool value) { return value ? "true" : "false"; }
+  template <unsigned BASE_BITS,
+            typename T,
+            typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
+  static std::string BaseConvert(T value, bool upper) {
+    std::string ret;
+    const char* digits = upper ? "0123456789ABCDEF" : "0123456789abcdef";
+    do {
+      unsigned digit = value & ((1 << BASE_BITS) - 1);
+      ret.push_back(
+          (BASE_BITS < 4 ? static_cast<char>('0' + digit) : digits[digit]));
+    } while ((value >>= BASE_BITS) != 0);
+    std::reverse(ret.begin(), ret.end());
```

Comment:
That's a very slow way of creating the string. I'd use a fixed-size array (`char ret[2 * sizeof(T)];`) and fill it back to front.